tuf/keydb.py

```diff
@@ -111,10 +117,21 @@ def create_keydb_from_root_metadata(root_metadata, repository_name='default'):
   else:
     create_keydb(repository_name)
 
-  # Iterate the keys found in 'root_metadata' by converting them to
-  # 'RSAKEY_SCHEMA' if their type is 'rsa', and then adding them to the
-  # key database using the provided keyid.
-  for keyid, key_metadata in six.iteritems(root_metadata['keys']):
+
+  keys_to_add = root_metadata['keys']
+
+  if targets_map_file is not None:
+    tuf.formats.TARGETS_MAPFILE_SCHEMA.check_match(targets_map_file)
+
+    # If any keyids from root metadata match those from targets_map_metadata,
+    # only the root metadata keys will be added. Add all keys to root_metadata
+    keys_to_add = dict(list(targets_map_file['keys'].items()) +
+        list(root_metadata['keys'].items()))
```

Comment:
Minor nit: This technique is discouraged (see e.g. https://stackoverflow.com/questions/38987/how-do-i-merge-two-dictionaries-in-a-single-expression-in-python-taking-union-o). I suggest to use the copy + update technique.